lib/pages/chat_detail_page.dart

```diff
@@ -0,0 +1,193 @@
+import 'package:clone_instagram/tests/test.dart';
+import 'package:clone_instagram/widgets/custom_chat_message_item.dart';
+import 'package:flutter/material.dart';
+
+import '../constants/app_text_style.dart';
+import '../constants/source_string.dart';
+import '../data/chat_message.dart';
+import '../widgets/custom_call_item_widget.dart';
+
+class ChatDetailPage extends StatefulWidget {
+  const ChatDetailPage({super.key});
+
+  @override
+  State<ChatDetailPage> createState() => _ChatDetailPageState();
+}
+
+class _ChatDetailPageState extends State<ChatDetailPage> {
+  bool isTyping = false;
+  String typingMessage = "";
+  late String contactorId;
+  late List<ChatMessage> messageList;
+
+  TextEditingController textEditingController = TextEditingController();
+
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    typingMessage = textEditingController.text;
+    contactorId = ModalRoute.of(context)!.settings.arguments as String;
+    messageList = Test.chatMessageList.reversed.where((message) {
+      return (message.senderId == "U000" && message.receiverId == contactorId) ||
+          (message.receiverId == "U000" && message.senderId == contactorId);
+    }).toList();
+
+    return GestureDetector(
```

Comment:
you should extract view into method such as _header, _footer, _body, _input, etc

Reply:
I changed.